catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CqlCommands.java

```diff
@@ -0,0 +1,238 @@
+                    .is()
+                    .during()
+                    .last(start);
+        } else {
+            if (caseSensitive) {
+                return filterBuilder.attribute(Metacard.ANY_TEXT)
+                        .is()
+                        .like()
+                        .caseSensitiveText(searchPhrase);
+            } else {
+                return filterBuilder.attribute(Metacard.ANY_TEXT)
+                        .is()
+                        .like()
+                        .text(searchPhrase);
+            }
+        }
+    }
+
+    protected long getFilterStartTime() {
+        if (lastSeconds > 0) {
+            return filterCurrentTime - TimeUnit.SECONDS.toMillis(lastSeconds);
+        } else if (lastMinutes > 0) {
+            return filterCurrentTime - TimeUnit.MINUTES.toMillis(lastMinutes);
+        } else if (lastHours > 0) {
+            return filterCurrentTime - TimeUnit.HOURS.toMillis(lastHours);
+        } else if (lastDays > 0) {
+            return filterCurrentTime - TimeUnit.DAYS.toMillis(lastDays);
+        } else if (lastWeeks > 0) {
+            Calendar weeks = GregorianCalendar.getInstance();
+            weeks.setTimeInMillis(filterCurrentTime);
+            weeks.add(Calendar.WEEK_OF_YEAR, -1 * lastWeeks);
+            return weeks.getTimeInMillis();
+        } else if (lastMonths > 0) {
+            Calendar months = GregorianCalendar.getInstance();
+            months.setTimeInMillis(filterCurrentTime);
+            months.add(Calendar.MONTH, -1 * lastMonths);
+            return months.getTimeInMillis();
+        } else {
+            return 0;
+        }
+    }
+
+    protected String getTemporalProperty() {
+        if (metacardTypes != null && StringUtils.isNotEmpty(temporalProperty)) {
+            return metacardTypes.stream()
+                    .map(metacardType -> metacardType.getAttributeDescriptors())
+                    .flatMap(Set::stream)
+                    .filter(isDateType())
+                    .map(AttributeDescriptor::getName)
+                    .filter(temporalProperty::equalsIgnoreCase)
+                    .findFirst()
+                    .orElse(DEFAULT_TEMPORAL_PROPERTY);
+        }
+
+        return DEFAULT_TEMPORAL_PROPERTY;
+    }
+
+    private Predicate<AttributeDescriptor> isDateType() {
+        return attributeDescriptor -> attributeDescriptor.getType()
+                .getAttributeFormat() == AttributeType.AttributeFormat.DATE;
+    }
+
+    protected Boolean hasFilter() {
+        return notNull(cqlFilter, startDate, endDate) || notZero(lastSeconds,
+                lastMinutes,
+                lastHours,
+                lastDays,
+                lastWeeks,
+                lastMonths);
+    }
+
+    private Boolean notNull(String... parameters) {
+        return Arrays.stream(parameters)
+                .anyMatch(p -> p != null);
+    }
+
+    private Boolean notZero(int... parameters) {
```

Comment:
It looks like this method will return TRUE if any parameter is >0. But the method name makes it sound like it return TRUE if all parameters are >0.